cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/model/impl/OracleGenerator.scala

```diff
@@ -0,0 +1,408 @@
+          val producedExprMap = producedExprs.result().toMap
+          val newExprMap =
+            producedExprMap.map {
+              case (field, expr) => field -> FieldExpr(s"$groupedLookupSourceVar.$quote$field$quote", s"$groupedLookupSourceVar.$quote$field$quote")
+            }
+          queryBuilder.insert(0, s"from (\n${parseProject(producedExprMap)}\n")
+          queryBuilder.append(s"\n) $groupedLookupSourceVar\n")
+          val resultAfterLookup = parseLookup(group.lookups, newExprMap, queryBuilder, true)
+          ParsedResult(Seq.empty, resultAfterLookup.exprMap)
+        } else {
+          ParsedResult(Seq.empty, producedExprs.result().toMap)
+        }
+
+      case None => ParsedResult(Seq(""), exprMap)
+    }
+  }
+
+  protected def parseCreate(create: CreateView, schemaMap: Map[String, AbstractSchema]): String = {
+
+    val (temporalSchemaMap, lookupSchemaMap) = GeneratorUtil.splitSchemaMap(schemaMap)
+    val sourceSchema = temporalSchemaMap(create.query.dataset)
+    val resultSchema = calcResultSchema(create.query, sourceSchema)
+    val ddl: String = genDDL(create.dataset, sourceSchema)
+    val primaryKeystr:Seq[String] = for (pk<- sourceSchema.primaryKey)yield{
+      pk.name
+    }
+    val measurementStrd:Seq[String] = for (d <- sourceSchema.measurement) yield {
+      "d.\""+d.name+"\""
+    }
+    val measurementStrs:Seq[String] = for (d <- sourceSchema.measurement) yield {
+      "s.\""+d.name+"\""
+    }
+    var dimensionStrd:Seq[String] = Seq()
+    for (d<- sourceSchema.dimension){
+      if (d.dataType != DataType.Hierarchy){
+        dimensionStrd= dimensionStrd :+"d.\"" + d.name + "\""
+      }
+    }
+    var dimensionStrs:Seq[String] = Seq()
+    for (d<- sourceSchema.dimension){
+      if (d.dataType == DataType.Hierarchy){
+      }
+      else {
+        dimensionStrs = dimensionStrs:+"s.\"" + d.name + "\""
+      }
+    }
+    var conflictsting = ""
+    for (pkname<- primaryKeystr){
+      conflictsting += "d.\""+pkname+"\" = s.\"" + pkname+"\" and "
+    }
+    val insert = s"""
+                    |merge into $quote${create.dataset}$quote d
+                    |using (${parseQuery(create.query,schemaMap)}) s
+                    |on (${conflictsting.dropRight(5)})
+                    |when not matched then
+                    |insert (${(dimensionStrd ++ measurementStrd).mkString(",")})
+                    |values (${(dimensionStrs ++ measurementStrs).mkString(",")})
+                    |""".stripMargin
+    ddl + insert
+  }
+
+  /**
+    * Convert middleware datatype to OracleSQL datatype
+    * @param field
+    */
+  protected def fieldType2SQLType(field: Field): String = {
+    field.dataType match {
+      case DataType.Number => "NUMBER"
+      case DataType.Time => "TIMESTAMP"
+      case DataType.Point => "SDO_GEOMETRY"
+      case DataType.Boolean => "NUMBER(1)"
+      case DataType.String => "VARCHAR2(255)"
+      case DataType.Text => "VARCHAR(1000)"
```

Comment:
`Text` in AsterixDB can NOT be matched a fixed length `VARCHAR` in Oracle, which will cause trouble if the length of values exceed the 1000 length which is not rare. We may need to consider `clob` in Oracle.